### DISCUSSION TITLE
i/o timeout when pulling image on Windows

### DISCUSSION THREAD

**Discussion by loicmathieu** [rancher-sandbox/rancher-desktop#1566] [Q&A] [answered]
Hi,

When I try to pull an image on Windows 10 I have an I/O timeout:
```
docker pull mongo:4.4
Error response from daemon: Get "https://registry-1.docker.io/v2/": dial tcp: lookup registry-1.docker.io on 172.28.80.1:53: read udp 172.28.84.59:35431->172.28.80.1:53: i/o timeout
```

I can run existing images.
Rancher desktop successfully pull its Kubernetes images.

My laptop is running on a corporate network so I have a proxy and maybe a custom DNS (this look like a DNS issue), I also have a VPN but the same issue arise with and without it so it may not be related to the VPN.

I try to login inside the rancher desktop WSL distribution to change the DNS but there is no `vi` in it so I cannot update the resolve.conf file.

Any suggestion ?

**Comment by adamkpickering** [comment]
As you say, it sounds like the VPN isn't the issue. And I would agree, from that error it looks like a DNS issue. One thing you could try is changing /etc/resolv.conf inside the VM. Windows isn't my strong suit though - @mook-as do you have any other ideas?

**Comment by mook-as** [comment]
I've also been having intermittent DNS issues in WSL2 sometimes; I have yet to figure out the issue (though it's mostly linked to slow VMs for me).

(We do have `busybox`, so `busybox vi` should work fine.)

**Comment by loicmathieu** [comment]
I update the DNS of the WSL distribution to use `8.8.8.8` but still have the same issue:

Here are the commands I use to update resolv.conf
```
busybox rm /etc/resolv.conf
busybox echo "nameserver 8.8.8.8" > /etc/resolv.conf
busybox echo "[network]" > /etc/wsl.conf
busybox echo "generateResolvConf = false" >> /etc/wsl.conf
```

**Reply by ssoft-wankun** [reply]
this helps,love you man

**Comment by jcageman** [comment]
exactly got the same issue and indeed changing resolv.conf this doesn't fix it :( It's very much related to the way it interacts with VPN, since if i disconnect the VPN it works again. I've installed https://github.com/sakai135/wsl-vpnkit and that seems to do the trick for me! See: https://github.com/microsoft/WSL/issues/4246

**Comment by jandubois** [comment, marked as answer]
Could you please try the new 1.1.1 release and report if this fixes your problem? It automatically switches the nameserver to the DNS nameserver when the VPN is connected.

**Reply by jcageman** [reply]
i've installed the 1.1.1 release, but still have the same issue with pulling images

```
nerdctl pull nginx
docker.io/library/nginx:latest: resolving      |--------------------------------------|
elapsed: 9.9 s                  total:   0.0 B (0.0 B/s)
INFO[0010] trying next host                              error="failed to do request: Head \"https://registry-1.docker.io/v2/library/nginx/manifests/latest\": net/http: TLS handshake timeout" host=registry-1.docker.io
FATA[0010] failed to resolve reference "docker.io/library/nginx:latest": failed to do request: Head "https://registry-1.docker.io/v2/library/nginx/manifests/latest": net/http: TLS handshake timeout
```

Anything i can test out on my end to find the root cause? I've connected to wsl and did ping www.google.com which now works fine with 1.1.1, but i still get these TLS handshake timeouts :(

**Comment by loicmathieu** [comment]
@jandubois version 1.1.1 fixes the issue, thanks !

**Comment by DevangJayswal** [comment]
I am also running rancher-desktop 1.1.1 in windows but still no luck. facing the same issue:

**Error response from daemon: Get "https://registry-1.docker.io/v2/": dial tcp: lookup registry-1.docker.io on 172.17.145.232:53: read udp 172.17.145.232:57991->172.17.145.232:53: i/o timeout**

**Reply by JavaGit2Hub** [reply]
I am NOT using kubernates and using dockerd(Moby) using v1.1.1 docker pull not working.(Even one of my friend enabled kubernates but docker pull not working)

**Reply by kasingal** [reply]
I am using Rancher Desktop v1.3.0 and docker login itself timing out :D

**Reply by ariolab** [reply]
I found out the solution that works for me.
`wsl -d rancher-desktop`
while in distro edit **/etc/resolv.conf** :
`options use-vc`
`nameserver 8.8.8.8`

Hope this helps to others facing the same issue.

**Comment by xiaods** [comment]
v1.3.3 rancher desktop
My laptop is running on a corporate network

I can't pull image from internal registry. it report timeout

**Comment by deMD** [comment]
Rancher Desktop 1.4.1 also experiencing this issue:
```
Error response from daemon: Get "https://mcr.microsoft.com/v2/": dial tcp: lookup mcr.microsoft.com on 172.31.36.177:53:
 read udp 172.31.36.177:39000->172.31.36.177:53: i/o timeout
```
